File: day 084/classwork/code/codewars1.js

// https://www.codewars.com/kata/52fba2a9adcd10b34300094c/train/javascript

// Matrix Transpose 

function transpose(matrix) {
  return matrix[0].map((_, colindex) => 
     matrix.map(row => row[colindex])                  
  );
}

File: day 084/homework/code/codewars1.py

# https://www.codewars.com/kata/5b817c2a0ce070ace8002be0/train/python

# Tic-Tac-Toe-like table Generator

def display_board(board, width):
    rows = []
    
    for i in range(0, len(board), width):
        row = "|".join(f" {cell} " for cell in board[i:i+width])
        rows.append(row)
    
    separator = "\n" + "-" * len(rows[0]) + "\n"
    
    return separator.join(rows)
